### DISCUSSION TITLE
React Native Auth + Database Boilerplate

### DISCUSSION THREAD

**Discussion by matsyui** [supabase/supabase#17524] [Feature Requests]
We've been impressed with the incredible work you've done with Supabase, and we greatly appreciate the value it brings to developers. We're reaching out to kindly request your assistance in creating a React Native Auth + Database Boilerplate using Supabase.

Our goal is to simplify the development process for React Native mobile applications by combining the power of React Native with the ease and security of Supabase. We believe that such a boilerplate would be immensely helpful to the developer community.

**Features We Envision:**
- **React Native Integration:** A solution that seamlessly integrates React Native for both iOS and Android platforms.
- **User Authentication:** Implementation of user authentication using Supabase for secure registration, login, and password reset functionality.
- **Database Integration:** Integration of Supabase as the database backend for efficient user data storage, retrieval, and management.
- **State Management:** We would appreciate the inclusion of either Zustand for state management.
- **User-Friendly UI Components:** Pre-designed UI components or theming options to facilitate the creation of a polished user interface.
- **Navigation:** An established navigation system that aligns with best practices in React Native app development.
- **User Profile:** A user profile screen where users can view and edit their profile information.
- **Documentation:** Clear and comprehensive documentation on setup, configuration, and usage of the boilerplate.

**Additional Notes:**
- We understand that you are doing this out of the goodness of your hearts, and we are extremely grateful for your contributions to the developer community. We want to emphasize that we've explored existing resources, including the examples provided in the Supabase GitHub repository at [https://github.com/supabase/supabase/tree/master/examples](https://github.com/supabase/supabase/tree/master/examples). However, it appears that most of the examples are tailored for Flutter and Next.js, which does not fully address our React Native needs. Additionally, we've searched for examples and tutorials from other sources, but most of them are deprecated or outdated by 2-3 years and are no longer maintained.
- Any support for social authentication methods (e.g., Google, Facebook) would be a valuable addition.
- Ensuring that the boilerplate is customizable and extensible is highly appreciated.

Thank you for considering our request.

**Comment by saltcod** [comment]
Hi there! Working on better RN materials as we speak. In the meantime, here's some recent RN content: 
https://supabase.com/blog/react-native-storage

and the repo for that:
https://github.com/saimon24/react-native-resumable-upload-supabase

**Comment by thorwebdev** [comment]
Thanks for the great suggestions @mashwishi, this is something I'm working towards. In the meantime, have you had a look at https://tamagui.dev/takeout ? I know this is a paid offering, so not ideal, but looks like it's ticking a lot of the boxes you mentioned. 

In general any thoughts on https://tamagui.dev/ ? Was thinking to use this for the boilerplate.

**Reply by matsyui** [reply]
Thanks for the offer, actually the offer is good but i think im good.

Been already starting actually, I just need a reference like a proper way implementing it, security and etc. in short the right flow of the system when implementing auth (login/register) + best way to request data from database with logged in user.

![image](https://github.com/supabase/supabase/assets/62344860/4f2c2be3-efa9-4337-826c-ab11cca3658c)

**Reply by thorwebdev** [reply]
Gotcha, I think the Expo User Management starter is a good example for that: https://supabase.com/docs/guides/getting-started/tutorials/with-expo-react-native

**Reply by matsyui** [reply]
Yup, i already use this and did abit different approach since i am using zustand.
